### PR TITLE
Add cnvkit genemetrics

### PR DESCRIPTION
**Pull request by grantn5**
## PR checklist

This PR adds the genemetrics module to CNVkit which creates an extra genemetrics file. Showing CNVs on a per gene level. If annotation is present, else an empty file is made

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Hi @grantn5,
We're about to release soon, so we'll probably merge your PR after, and it'll be available for the next one, which will be fairly soon as well

**Comment by FriederikeHanssen** [comment]
Hey! Thanks for adding this. Looks like you might not be quite up to date on the dev branch, the Changelog is missing quite some things. Can you merge in dev?

**Comment by grantn5** [comment]
> Hey! Thanks for adding this. Looks like you might not be quite up to date on the dev branch, the Changelog is missing quite some things. Can you merge in dev?

HI That should be done

File: modules/nf-core/cnvkit/genemetrics/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        'https://depot.galaxyproject.org/singularity/cnvkit:0.9.9--pyhdfd78af_0':
+        'quay.io/biocontainers/cnvkit:0.9.9--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(cnr)
+    tuple val (meta), path(cns)
+
+    output:
+    tuple val(meta), path("*.tsv"), emit: tsv
+    //tuple val(meta), path("*.cnn"), emit: cnn
```
Comment (FriederikeHanssen) [inline modules/nf-core/cnvkit/genemetrics/main.nf:16]: I know this is from upstream, but any idea what is going on here?

File: modules/nf-core/cnvkit/genemetrics/main.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
+process CNVKIT_GENEMETRICS {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::cnvkit=0.9.9 bioconda::samtools=1.16.1"
```
Comment (FriederikeHanssen) [inline modules/nf-core/cnvkit/genemetrics/main.nf:5]: There should be a new container available with 0.9.10 and samtools 1.17 that we use for batch now.

File: subworkflows/local/bam_variant_calling_cnvkit/main.nf [hunk 3]
```diff
@@ -18,6 +19,7 @@ workflow BAM_VARIANT_CALLING_CNVKIT {
     generate_pon = false
 
     CNVKIT_BATCH(cram, fasta, fasta_fai, targets, reference, generate_pon)
+    CNVKIT_GENEMETRICS(CNVKIT_BATCH.out.cnr, CNVKIT_BATCH.out.cns)
 
     versions = CNVKIT_BATCH.out.versions
```
Comment (FriederikeHanssen) [inline subworkflows/local/bam_variant_calling_cnvkit/main.nf:24]: can you add the version collection for the new module?

File: modules/nf-core/cnvkit/genemetrics/main.nf (hunk 1 above)
Comment (grantn5) [reply modules/nf-core/cnvkit/genemetrics/main.nf:16]: As If i left it as emitting the `.cnn` it overwirtes some other outputs so I changed it to .tsv

Comment (FriederikeHanssen) [reply modules/nf-core/cnvkit/genemetrics/main.nf:16]: You could change the output name using `ext.prefix` and avoid any overwritting.

File: CHANGELOG.md [hunk 4]
```diff
@@ -63,6 +63,7 @@ Rapaselet is a delta formed by the Rapaätno river between the Bielloriehppe mas
 - [#1173](https://github.com/nf-core/sarek/pull/1173) - CI tests for VQSR track with stub runs
 - [#1122](https://github.com/nf-core/sarek/pull/1122), [#1196](https://github.com/nf-core/sarek/pull/1196) - Add `annotation cache` functionality
 - [#1184](https://github.com/nf-core/sarek/pull/1184) - Stub-based CI-test of Sentieon joint-germline variant-calling with VQSR
+- [#1113](https://github.com/nf-core/sarek/pull/1113) - Adding CNVkit genemetrics module
```
Comment (maxulysse) [inline CHANGELOG.md:66]: can you move this to the `dev` part?

File: modules/nf-core/cnvkit/genemetrics/main.nf (hunk 1 above)
Comment (grantn5) [reply modules/nf-core/cnvkit/genemetrics/main.nf:16]: Although it does output a tsv file so makes it easier for a user to know what delimiter to use when loading into r/python

**Comment by FriederikeHanssen** [comment]
Hey! Sorry for the delay once this PR is merged the tests should be triggered automatically. Did you test the genemetrics implementation on some data already? I am just wondering if the gene name display works properly, because we currently still have a feature request open to the refFlat file.

**Comment by FriederikeHanssen** [comment]
@nf-core-bot  fix linting

**Comment by grantn5** [comment]
> Hey! Sorry for the delay once this PR is merged the tests should be triggered automatically. Did you test the genemetrics implementation on some data already? I am just wondering if the gene name display works properly, because we currently still have a feature request open to the refFlat file.

Hi Yes I have tested that it works on some of my own data, i usually annotate my data with a bed file from ensembl and it the names appear fine

**Comment by FriederikeHanssen** [comment]
@nf-core-bot fix linting

**Comment by FriederikeHanssen** [comment]
@maxulysse can you review, Grant and me pair-programmed the PR now :D

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## dev
 
+- [#1113](https://github.com/nf-core/sarek/pull/1113) - Adding CNVkit genemetrics module
+- [#1246](https://github.com/nf-core/sarek/pull/1246) - Back to dev
```
Comment (asp8200) [inline CHANGELOG.md:11]: I don't think that old PR #1246 should be listed there.

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## dev
 
+- [#1113](https://github.com/nf-core/sarek/pull/1113) - Adding CNVkit genemetrics module
```
Comment (asp8200) [inline CHANGELOG.md:10]: Should that be listed under `### Added`?

File: docs/output.md [hunk 7]
```diff
@@ -717,8 +717,9 @@ The file `<tumorsample_vs_normalsample>.cnvs.txt` contains all segments predicte
   - file containing copy number segment information
 - `<sample>.call.cns`
   - file containing copy number segment information
-
-</details>
+- `<sample>.genemetrics.tsv`
+  - file containing per gene copy number information (if input files are annotated)
+  </details>
```
Comment (maxulysse) [inline docs/output.md:722]:
```suggestion
</details>
```

**Comment by maxulysse** [comment]
> @maxulysse can you review, Grant and me pair-programmed the PR now :D

Looking good, I think we're just missing the commented line in the output block

**Pull request merged**